include/alpaka/math/min/MinStdLib.hpp

```diff
@@ -27,32 +27,21 @@ namespace alpaka
 
         namespace traits
         {
-            //! The standard library integral min trait specialization.
+            //! The standard library min trait specialization.
             template<typename Tx, typename Ty>
-            struct Min<MinStdLib, Tx, Ty, std::enable_if_t<std::is_integral<Tx>::value && std::is_integral<Ty>::value>>
-            {
-                ALPAKA_FN_HOST auto operator()(MinStdLib const& min_ctx, Tx const& x, Ty const& y)
-                {
-                    alpaka::ignore_unused(min_ctx);
-                    using std::min;
-                    return min(x, y);
-                }
-            };
-            //! The standard library mixed integral floating point min trait specialization.
-            template<typename Tx, typename Ty>
-            struct Min<
-                MinStdLib,
-                Tx,
-                Ty,
-                std::enable_if_t<
-                    std::is_arithmetic<Tx>::value && std::is_arithmetic<Ty>::value
-                    && !(std::is_integral<Tx>::value && std::is_integral<Ty>::value)>>
+            struct Min<MinStdLib, Tx, Ty, std::enable_if_t<std::is_arithmetic_v<Tx> && std::is_arithmetic_v<Ty>>>
             {
                 ALPAKA_FN_HOST auto operator()(MinStdLib const& min_ctx, Tx const& x, Ty const& y)
                 {
                     alpaka::ignore_unused(min_ctx);
+
                     using std::fmin;
-                    return fmin(x, y);
+                    using std::min;
+
+                    if constexpr(std::is_integral_v<Tx> && std::is_integral_v<Ty>)
+                        return min(x, y);
+                    else // Floating point or mixed integral-floating point
```

Comment:
This part should be explicit too and the else branch should contain a static assert.